lib/src/domain/usecase/usecase_interface.dart

```diff
@@ -0,0 +1,7 @@
+abstract class IUseCaseMovie<T> {
+  Future<T> repositoryCall();
+}
+
+abstract class IUseCaseGenres<T> {
+  Future<T> repositoryCall();
+}
```

Comment:
This is the same right? You can have only one abstract class that define the call()

Reply:
You are right, I changed to have only one interface "IUseCase" for both genres and movies